src/MF_Analog/MFAnalog.cpp

```diff
@@ -15,20 +15,32 @@ MFAnalog::MFAnalog(uint8_t pin, const char *name, uint8_t sensitivity)
     _name        = name;
     _lastValue   = 0;
     pinMode(_pin, INPUT_PULLUP); // set pin to input. Could use OUTPUT for analog, but shows the intention :-)
-    analogRead(_pin);            // turn on pullup resistors
+    for (uint8_t i = 0; i < ADC_MAX_AVERAGE; i++) {
+        readBuffer();
+    }
 }
 
-void MFAnalog::update()
+void MFAnalog::readChannel(uint8_t compare)
```

Comment:
the parameter name `compare` could be more explicitly describing what we mean so that it becomes easier to understand when reading:

`forceReadandAlwaysTrigger`
`alwaysTrigger`

And also 
```abs(newValue - _lastValue) >= _sensitivity```

would be nicer in a short function `valueHasChanged`